src/controllers/events.js

```diff
@@ -29,3 +30,16 @@ exports.getRelatedEvents = async (req, res, next) => {
     .populate('ngo', 'name');
   return res.status(200).json({ success: true, data: relatedEvents });
 };
+
+exports.addEvent = async (req, res, next) => {
+  // Create new event
+  const newEvent = await Event.create(req.body);
+  newEvent.ngo = req.user._id;
+  await newEvent.save();
+  //  Add event to the ngo
+  console.log(req.user.type);
+  const ngo = await Ngo.findById(req.user._id, {});
+  ngo.publishedEvents.push(newEvent._id);
+  await ngo.save();
```

Comment:
Although, this works and totally safe. Let me give you another optimized and short alternative:

```javascript
await Ngo.updateOne({_id: req.user._id}, { $push: { publishedEvents: newEvent._id } })
```